### PR TITLE
SOF-5356 - New espresso version tests

### PR DESCRIPTION
**Pull request by AcylSilane**

### CODE REVIEW

File: express/parsers/apps/espresso/parser.py [hunk 1]
```diff
@@ -356,14 +356,30 @@ def _is_sternheimer_gw_calculation(self):
...
+        files = []
+        directories = []
+        for item in map(lambda i: os.path.join(cwd, i), os.listdir(cwd)):
+            if os.path.isdir(item):
+                directories.append(item)
+            elif os.path.isfile(item):
+                files.append(item)
+
+        for file in files:
+            if file.endswith("xml"):
```
Comment (timurbazhirov) [inline express/parsers/apps/espresso/parser.py:378]: 1. This one will only ever find one XML file - is there a specific naming convention for it? This coulbe be prone to error if a workflow has multiple steps reusing the same directory. 

2. How did `find_xml_file()` work before - why did we need to adjust this?

Comment (AcylSilane) [reply express/parsers/apps/espresso/parser.py:378]: Figured out the failure! What was going on is that the settings.XML_DATA_FILE variable is set to `data-file.xml`. The new versions of Espresso store this as `data-file-schema.xml`. Fixed this by storing it as an attribute of the different parsers, for now.

File: express/parsers/apps/espresso/parser.py [hunk 2]
```diff
@@ -346,6 +347,10 @@ def final_structure_strings(self) -> List[str]:
 
 class EspressoParser(EspressoLegacyParser):
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.xml_data_file = "data-file-schema.xml"
```
Comment (timurbazhirov) [inline express/parsers/apps/espresso/parser.py:352]: `xml_data_file_name`

File: tests/__init__.py [hunk 3]
```diff
@@ -1,8 +1,19 @@
+import functools
 import os
 import yaml
 import unittest
 import numpy as np
 
+def for_all_versions(version_map: dict, runtype_map: dict):
```
Comment (timurbazhirov) [inline tests/__init__.py:7]: Let's add a docstring above

File: express/parsers/apps/espresso/parser.py (hunk 2 above)
Comment (AcylSilane) [reply express/parsers/apps/espresso/parser.py:352]: Fixed, and refactored the `find_xml_file` method above for readability

**Comment by AcylSilane** [comment]
Closing because 5394 contains the new test work

**Pull request closed without merging**